### PR TITLE
matt/bug_fixes

### PR DESCRIPTION
**Pull request by mattbarker016**

### CODE REVIEW

File: TCAT/Controllers/RouteDetailViewController.swift [hunk 1]
```diff
@@ -95,7 +95,7 @@ class RouteDetailViewController: UIViewController, GMSMapViewDelegate, CLLocatio
                 
                 var waypoints: [Waypoint] = []
                 for point in path {
-                    var type: WaypointType = .none
+                    var type: WaypointType = .walking
                     if direction == self.directions.first && point == path.first! {
```
Comment (AAAstorga) [inline TCAT/Controllers/RouteDetailViewController.swift:99]: Don't assume first exists. do an if let please

File: TCAT/Controllers/RouteDetailViewController.swift [hunk 2]
```diff
@@ -413,15 +397,16 @@ class RouteDetailViewController: UIViewController, GMSMapViewDelegate, CLLocatio
 
                 UIView.animate(withDuration: 1.0, delay: 0, options: .curveEaseInOut, animations: {
                     newBus.userData = bus
-                    (newBus.iconView as? BusLocationView)?.setBearing(bus.heading, start: existingBus!.position, end: busCoords)
+                    // (newBus.iconView as? BusLocationView)?.setBearing(bus.heading, start: existingBus!.position, end: busCoords)
```
Comment (AAAstorga) [inline TCAT/Controllers/RouteDetailViewController.swift:400]: can you delete this?

File: TCAT/Controllers/RouteDetailViewController.swift [hunk 3]
```diff
@@ -458,6 +443,21 @@ class RouteDetailViewController: UIViewController, GMSMapViewDelegate, CLLocatio
...
+    func setIndex(of marker: GMSMarker, with waypoint: Waypoint) {
+        marker.zIndex = {
+            switch waypoint.wpType {
+            case .bus: return 1
+            case .walk: return 1
+            case .origin: return 3
+            case .destination: return 3
+            case .stop: return 1
+            case .walking: return 0
+            case .bussing: return 999
```
Comment (AAAstorga) [inline TCAT/Controllers/RouteDetailViewController.swift]: explain why you return 999 please

**Review by AAAstorga — APPROVED** [review]
after doing the fixes, merge it on in

File: TCAT/Controllers/RouteDetailViewController.swift (hunk 2 above)
Comment (mattbarker016) [reply TCAT/Controllers/RouteDetailViewController.swift:400]: I toggle between `bearing` and `betterBearing`, not sure why you don't have` betterBearing`

File: TCAT/Controllers/RouteDetailViewController.swift (hunk 3 above)
Comment (mattbarker016) [reply TCAT/Controllers/RouteDetailViewController.swift]: The random high number worked, it's just a high constant

File: TCAT/Controllers/RouteDetailViewController.swift (hunk 1 above)
Comment (mattbarker016) [reply TCAT/Controllers/RouteDetailViewController.swift:99]: If the for loop fires, there must be at least one element. That's how you get EXCITED in Swift properly

**Pull request merged**